internal/command/jsonformat/computed/renderers/object.go

```diff
@@ -36,11 +36,15 @@ type objectRenderer struct {
 
 func (renderer objectRenderer) RenderHuman(diff computed.Diff, indent int, opts computed.RenderHumanOpts) string {
 	if len(renderer.attributes) == 0 {
-		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(diff.Replace))
+		return fmt.Sprintf("{}%s%s", nullSuffix(opts.OverrideNullSuffix, diff.Action), forcesReplacement(diff.Replace, opts.OverrideForcesReplacement))
 	}
 
 	attributeOpts := opts.Clone()
 	attributeOpts.OverrideNullSuffix = renderer.overrideNullSuffix
+	// If the object is in a set then there is a good chance it has been told
+	// to override forces replacement, but we know that the children of objects
+	// never need to do this so we just set this false explicitly regardless.
```

Comment:
Why do we know that the children of objects never need to do this? I assumed (possibly incorrectly) that the protocol would allow arbitrary paths to be specified as causing the replacement, even if those paths traverse into an object attribute.

Reply:
Children of objects will still have their own internal Replace logic set by the protocol, but children of objects never need to override their internal logic. It's only nested sets and set blocks that will tell their children to override their internal replace boolean.

I've actually thought of a better way to implement this (which is to update the opts object itself to not propagate the override replace setting), which will mean I don't need to do this here or have a confusing comment.